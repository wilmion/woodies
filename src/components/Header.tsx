import React from 'react'
import { FiMenu } from 'react-icons/fi'

//IMAGES
import Logo from '../assets/images/ico/logo.svg'

import "../assets/scss/components/header.scss";

interface IProps {
    toggleMenu:(open:boolean) => void
}

const Header:React.FC<IProps> = ({toggleMenu}) => { 
    return (
        <header className="header">
            <div className="header__menu">
                <FiMenu 
                    className="header__menu__icon" 
                    onClick={() => toggleMenu(true)} 
                />
            </div>
            <nav className="header-nav" id="menu_mobile">
                <div className="header-nav__loge">
                    <picture className="header-nav__loge__img">
                        <img src={Logo} alt="Logo of woodies" />
                    </picture>
                    <h1 className="header-nav__loge__title">WOODIES</h1>
                </div>
                <ul className="header-nav-ul">
                    <li className="header-nav-ul-item">
                        <a href="/" className="header-nav-ul-item__link">Home</a>
                    </li>
                    <li className="header-nav-ul-item">
                        <a href="#about" className="header-nav-ul-item__link">About Us</a>
                    </li>
                    <li className="header-nav-ul-item">
                        <a href="#howitworks" className="header-nav-ul-item__link">How it works</a>
                    </li>
                    <li className="header-nav-ul-item">
                        <a href="#categories" className="header-nav-ul-item__link">Categories</a>
                    </li>
                    <li className="header-nav-ul-item">
                        <a href="#testimony" className="header-nav-ul-item__link">Testimony</a>
                    </li>
                    <li className="header-nav-ul-item">
                        <a href="/login" className="header-nav-ul-item__signUp">Sing Up</a>
                    </li>
                    <li 
                        onClick={() => toggleMenu(false)} 
                        className="header-nav-ul-item header-nav-ul-item--exit"
                    >Exit Menu</li>
                </ul>
            </nav>
        </header>
    )
}

export default Header
