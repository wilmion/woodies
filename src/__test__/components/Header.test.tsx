import React from 'react'
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header'; 

describe('<Header />' , () => {
    const func = jest.fn();
    const header = shallow(<Header toggleMenu={func} />);
    test('Render' , () => {
        const headerT = shallow(<Header toggleMenu={(open) => null} />);
        expect(headerT.length).toEqual(1);
    })
    test('Click on menu' , () => {    
        header.find('.header__menu__icon').simulate('click')
        expect(func).toBeCalledTimes(1)
    })
    test('Contain 3 items ' , () => {
        expect(header.find('.header-nav-ul').children().length).toEqual(7);
    })
    test('Snapshopt' , () => {
        const snapshot = create(<Header toggleMenu={jest.fn()} />).toJSON();

        expect(snapshot).toMatchSnapshot();
    })
})