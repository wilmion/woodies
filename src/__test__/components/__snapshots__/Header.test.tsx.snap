// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshopt 1`] = `
<header
  className="header"
>
  <div
    className="header__menu"
  >
    <svg
      className="header__menu__icon"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <nav
    className="header-nav"
    id="menu_mobile"
  >
    <div
      className="header-nav__loge"
    >
      <picture
        className="header-nav__loge__img"
      >
        <img
          alt="Logo of woodies"
          src={Object {}}
        />
      </picture>
      <h1
        className="header-nav__loge__title"
      >
        WOODIES
      </h1>
    </div>
    <ul
      className="header-nav-ul"
    >
      <li
        className="header-nav-ul-item"
      >
        <a
          className="header-nav-ul-item__link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="header-nav-ul-item"
      >
        <a
          className="header-nav-ul-item__link"
          href="#about"
        >
          About Us
        </a>
      </li>
      <li
        className="header-nav-ul-item"
      >
        <a
          className="header-nav-ul-item__link"
          href="#howitworks"
        >
          How it works
        </a>
      </li>
      <li
        className="header-nav-ul-item"
      >
        <a
          className="header-nav-ul-item__link"
          href="#categories"
        >
          Categories
        </a>
      </li>
      <li
        className="header-nav-ul-item"
      >
        <a
          className="header-nav-ul-item__link"
          href="#testimony"
        >
          Testimony
        </a>
      </li>
      <li
        className="header-nav-ul-item"
      >
        <a
          className="header-nav-ul-item__signUp"
          href="/login"
        >
          Sing Up
        </a>
      </li>
      <li
        className="header-nav-ul-item header-nav-ul-item--exit"
        onClick={[Function]}
      >
        Exit Menu
      </li>
    </ul>
  </nav>
</header>
`;
